packages/monitoring/src/serviceMonitor/alivenessChecker.ts

```diff
@@ -77,3 +92,209 @@ export class ServiceMonitor {
+ * @property {string} [mnemonic] - Mnemonic required for RMB service.
+ * @property {KeypairType} keypairType - Type of keypair, default is "sr25519".
+ * @property {boolean} rmbValidatesChain - Indicates if RMB will validate the chain url.
+ * @property {string[]} rmbTFchainUrls - URLs of TFChain for RMB service, Required in case there is no TFChain urls for tfChain service.
+ * If there is a tfChain availability service provided, and it's result is valid, we will ignore the rmb-tfChain options
+ * @property {boolean} silent - To return null instead of throw an Error, Default is False
+ */
+export class ServiceUrlManager<N extends boolean = false> {
+  private result: ServicesUrls<N> = {};
+  private retries = 3;
+  private [ServiceName.tfChain]?: string[];
+  private [ServiceName.GraphQl]?: string[];
+  private [ServiceName.RMB]?: string[];
+  private [ServiceName.GridPrixy]?: string[];
+  private [ServiceName.Activation]?: string[];
+  private [ServiceName.Stats]?: string[];
+  private mnemonic?: string;
+  private keypairType: KeypairType = "sr25519";
+  private rmbValidatesChain = false;
+  private rmbTFchainUrls: string[];
+  private silent: N = false as N;
+
+  constructor(options?: StackPickerOptions<N>) {
+    Object.assign(this, options);
+  }
+
+  /**
+   * Pings the given service to check if it is alive.
+   *
+   * This method checks the liveness of the provided service by calling its `isAlive` method.
+   * If the service supports disconnection (implements IDisconnectHandler), it calls the `disconnect` method after the liveness check.
+   *
+   * @param {ILivenessChecker} service - An instance of ILivenessChecker that provides methods to check the service's liveness.
+   *
+   * @returns {Promise<{alive: boolean, error?: Error}>} - A promise that resolves with the liveness status of the service.
+   */
+  private async pingService(service: ILivenessChecker) {
+    const status = await service.isAlive();
+    if ("disconnect" in service) {
+      await (service as IDisconnectHandler).disconnect();
+    }
+    return status;
+  }
+
+  /**
+   * Attempts to find a reachable service URL from a list of provided URLs.
+   *
+   * This method iterates through the list of URLs, repeatedly pinging the service to
+   * check if it is alive. If a reachable URL is found, it is returned. If all URLs
+   * are exhausted without finding a reachable service, an error is thrown.
+   *
+   * @param {string[]} urls - An array of service URLs to check for reachability.
+   * @param {ILivenessChecker} service - An instance of ILivenessChecker that provides methods
+   * to check the service's liveness and manage service URLs.
+   *
+   * @returns {Promise<string>} - A promise that resolves with the reachable service URL.
+   *
+   * @throws {Error} - Throws an error if no reachable service URL is found after checking all URLs.
+   *
+   */
+  async getAvailableServiceStack(urls: string[], service: ILivenessChecker): Promise<ServiceUrl<N>> {
```

Comment:
Here we should pass service instance, i don't like it but this will reduce code redundancy and  handle all types of ILivenessChecker